cogsmisc/customization.py

```diff
@@ -376,8 +379,62 @@ async def rename(self, ctx, old_name, new_name):
             await update_meth(ctx, sub_doc)
             return await ctx.send(
-                f"Okay, the workshop {self.obj_name} that was bound to {old_name} is now bound to {new_name}."
-            )
+                f"Okay, the workshop {self.obj_name} that was bound to {old_name} is now bound to {new_name}.")
+
+    async def serve(self, ctx, name):
+        if self.before_edit_check:
+            self.before_edit_check(ctx, name)
+        if not _can_edit_servaliases(ctx):
+            raise NotAllowed("You do not have permission to edit server aliases. Either __Administrator__ "
+                             "Discord permissions or a role named \"Server Aliaser\" or \"Dragonspeaker\" "
+                             "is required.")
+        if self.is_server:
+            return ctx.send(f"You can only serve personal {'aliases' if self.is_alias else 'snippets'}.")
+
+        # list of (name, (alias or sub doc, collection or None))
+        choices = []
+        if personal_obj := await self.personal_cls.get_named(name, ctx):
+            choices.append((f"{name} ({self.obj_name})",
+                            (personal_obj, None)))
+
+        # get list of (subscription object ids, subscription doc)
+        async for subscription_doc in self.workshop_sub_meth(ctx):
+            the_collection = await workshop.WorkshopCollection.from_id(ctx, subscription_doc['object_id'])
+            for binding in subscription_doc[self.binding_key]:
+                if binding['name'] == name:
+                    choices.append((f"{name} ({the_collection.name})",
+                                    (subscription_doc, the_collection)))
+
+        personal_obj, collection = await get_selection(ctx, choices)
+        embed = None
+        if isinstance(personal_obj, self.personal_cls):
+            if self.is_alias:
+                server_obj = personal.Servalias.new(personal_obj.name, personal_obj.code, ctx.guild.id)
+            else:
+                server_obj = personal.Servsnippet.new(personal_obj.name, personal_obj.code, ctx.guild.id)
+        else:  # We're working with a workshop collection, easy peasy
+            personal_obj.set_server_active(ctx)
+            embed = EmbedWithAuthor(ctx)
+            embed.title = f"Subscribed to {the_collection.name}"
+            embed.url = the_collection.url
+            embed.description = the_collection.description
+            if the_collection.aliases:
+                embed.add_field(name="Server Aliases" if self.is_server else "Aliases",
+                                value=", ".join(sorted(a.name for a in the_collection.aliases)))
+            if the_collection.snippets:
+                embed.add_field(name="Server Snippets" if self.is_server else "Snippets",
+                                value=", ".join(sorted(a.name for a in the_collection.snippets)))
+            return ctx.send(embed=embed)
+
+        await server_obj.commit(ctx.bot.mdb)
```

Comment:
while `server_obj` is guaranteed to be defined here because of the return, this code flow is difficult to follow